christfellowshipchurch/web-app
# [Events] Re-integrate Schedule data, refine Layout and Chat design

# Changes or Updates

**Schedule**
- Re-integrates the `<EventSchedule>` and Event calls to action data
- Breaks out the `<EventSchedule>` component into sub-component files for organizational clarity

**Chat**
- Adds an "offline" or not-live state for `<EventChat>` so chat code is not initialized at all for `!isLive` events
- Updates the chat `<MessageInput>` look and feel to match CF branding
- Disables/hides file uploading and emoji picker buttons (including sneaky drag-and-drop behavior)
- Updates typography and design of `<Message>` component to match CF branding

**Layout/General**
- Swaps order of tabs, so it goes **Schedule** | **Chat**
- Sets default tab based on `isLive` — when live, default tab is **Chat,** otherwise **Schedule**
- Theme value refinements around shadows etc
- Tweaks overall layout padding/design to make it more production ready

# Tradeoffs/Caveat
The actual schedule data and display is a bit clumsy, given the impeding API work that affects Event schema. This PR focused more on the _data_ integration and technical plumbing to get the data on the page — not polishing the design details, which will be a lower effort now.

Tablet/mobile responsiveness is _better_, but still less than ideal. Will revisit that shortly.

# Screenshots
|Caption|Screenshot/Gif|
|---|---|
|Default tab depends on `isLive` state|<img src="https://user-images.githubusercontent.com/1812955/94051595-bc5f1300-fda5-11ea-96ed-de7f5d42711e.gif" width="460" />|
|Updated Chat styles and preventing file uploads|<img src="https://user-images.githubusercontent.com/1812955/94051676-def12c00-fda5-11ea-8483-92d8e363ca5c.gif" width="460" />|
|Slightly cleaner "small but not quite tablet" size layout|<img src="https://user-images.githubusercontent.com/1812955/94051721-f16b6580-fda5-11ea-8ff1-93ef290419dc.jpg" width="460" />|
|Event Schedule re-integration (GIF)|<img src="https://user-images.githubusercontent.com/1812955/94051761-0516cc00-fda6-11ea-81f4-5165c16d2bc0.gif" width="460" />|
|Event Schedule screenshot|<img src="https://user-images.githubusercontent.com/1812955/94051841-25468b00-fda6-11ea-8f33-4ed8896ef54a.jpg" width="460" />|
|Event Schedule with "corrupt"/weird data (that will be changed soon)|<img src="https://user-images.githubusercontent.com/1812955/94051811-18c23280-fda6-11ea-88e9-160d67384190.jpg" width="460" />|

# Test Instructions
1. Open http://local.christfellowship.church:3000 and explore different events via the hamburger menu > Events page
1. 🔸 Verify events that are **not**  live default to the Schedule tab, and ones that **are live** default to Chat tab
1. 🔸 Verify the tab order is now **Schedule** then **Chat**
1. 🔸 Verify you cannot see the emoji picker or file upload in the Chat message input area, and that you cannot drag and drop files onto it to attach them
1. 🔸 Verify that the Schedule tab _generally_ works, even if it doesn't look great all the time or sometimes has clumsy data

# Tests
